crates/ruff/src/rules/flake8_pyi/rules/type_comment_in_stub.rs

```diff
@@ -0,0 +1,63 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+use rustpython_parser::lexer::LexResult;
+use rustpython_parser::Tok;
+
+use ruff_macros::{define_violation, derive_message_formats};
+
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+use crate::Range;
+
+define_violation!(
+    /// ## What it does
+    /// Do not use type comments (e.g. `x = 1 # type: int`) in stubs, even if
+    /// the stub supports Python 2. Always use annotations instead (e.g.
+    /// `x: int = 1`).
```

Comment:
I think this paragraph would make sense in the `## Why is it bad?` section, since `## What it does?` tends to be a description of what it's checking rather than alternatives etc. So e.g.:

```rs
/// ## What it does
/// Checks for the use of type comments (e.g., `x = 1  # type: int`) in stub
/// files.
///
/// ## Why is this bad?
/// Stub (`.pyi`) files should always use type annotations directly, rather
/// than type comments, even if they're intended to support Python 2, since
/// stub files are not executed at runtime.
///
/// ## Example
/// ```python
/// x = 1 # type: int
/// ```
///
/// Use instead:
/// ```python
/// x: int = 1
/// ```
```

Reply:
thanks!

Reply:
i've also added the `# type: ignore` exception